ui/src/time_picker.tsx

```diff
@@ -194,9 +203,15 @@ export const
         to ${formatDateToTimeString(parseTimeStringToDate(max), hour_format)}.`
 
     React.useEffect(() => {
-      if (format) wave.args[m.name] = value ? formatDateToTimeString(value, '24') : null
+      const
+        time = m.value ? parseTimeStringToDate(m.value) : null,
+        minTime = min ? parseTimeStringToDate(min) : null,
+        maxTime = max ? parseTimeStringToDate(max) : null,
```

Comment:
It would be better to precompute the parsedMin/Max just once instead of doing so in render, outOfBounds, here and getErrorMessage.